src/de.hpi.swa.graal.squeak.test/src/de/hpi/swa/graal/squeak/test/AbstractSqueakTestCaseWithImage.java

```diff
@@ -166,15 +166,12 @@ private static void resetSemaphoreLists() {
         resetList(NilObject.SINGLETON, interruptSema, "Interrupt semaphore");
         // The timer semaphore is taken care of in ensureTimerLoop, since the delays need to be
         // reset as well
+        final ArrayObject oldExternalObjects = (ArrayObject) image.getSpecialObject(SPECIAL_OBJECT.EXTERNAL_OBJECTS_ARRAY);
+        image.evaluate("[ ExternalObjectTable current " +
+                        "            initializeCaches;" +
+                        "            externalObjectsArray: (Smalltalk specialObjectsArray at: 39 put: (Array new: 20)) ] value");
```

Comment:
Why not `ExternalObjectTable reset`?

Reply:
because we also want to clean up the existing external semaphores registered in the previous (now dead) execution

Reply:
I mean the non-cached ones, inside the specialObjectsArray